mobile/native/src/max_quantity.rs

```diff
@@ -33,16 +35,75 @@ pub fn max_quantity(
         }
     };
 
-    let max_coordinator_margin =
+    let max_coordinator_balance =
         Amount::from_sat(channel_trade_constraints.max_counterparty_balance_sats);
-    let max_trader_margin = Amount::from_sat(channel_trade_constraints.max_local_balance_sats);
+    let max_trader_balance = Amount::from_sat(channel_trade_constraints.max_local_balance_sats);
     let order_matching_fee_rate = channel_trade_constraints.order_matching_fee_rate;
-    let order_matching_fee_rate =
-        Decimal::try_from(order_matching_fee_rate).expect("to fit into decimal");
+    let order_matching_fee_rate = Decimal::try_from(order_matching_fee_rate).expect("to fit");
 
-    let accumulated_order_matching_fees = match position::handler::get_positions()?.first() {
-        Some(position) => position.order_matching_fees,
-        _ => Amount::ZERO,
+    let positions = position::handler::get_positions()?;
+    let position = positions.first();
+    let accumulated_order_matching_fees = position
+        .map(|p| p.order_matching_fees)
+        .unwrap_or(Amount::ZERO);
+
+    let (max_coordinator_margin, max_trader_margin) = match position {
+        Some(position) if trader_direction != position.direction => {
+            let total_collateral = channel_trade_constraints
+                .total_collateral
+                .unwrap_or(Amount::ZERO);
+
+            let total_balance = channel_trade_constraints.max_counterparty_balance_sats
+                + channel_trade_constraints.max_local_balance_sats;
+            let total_balance = Amount::from_sat(total_balance);
+
+            let trader_margin = Amount::from_sat(position.collateral);
+            let counterparty_margin = total_collateral - (total_balance + trader_margin);
```

Comment:
🔧 Probably needs to be `checked_sub`bed, given that `total_collateral` can be zero.

Reply:
Maybe audit the PR diff for places where we should use `checked_sub` in general.